res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -1417,17 +1416,15 @@ PioneerDDJSX.tempoResetButton = function(channel, control, value, status, group)
 };
 
 PioneerDDJSX.autoLoopButton = function(channel, control, value, status, group) {
-    if (value) {
-        if (engine.getValue(group, "loop_enabled")) {
-            engine.setValue(group, "reloop_toggle", true);
-        } else {
-            engine.setValue(group, "beatloop_activate", true);
-        }
+    if (engine.getValue(group, "loop_enabled")) {
+        engine.setValue(group, "reloop_toggle", value);
+    } else {
+        engine.setValue(group, "beatloop_activate", value);
```

Comment:
This will not work as expected. Setting beatloop_activate to 1 sets loop_enabled to 1, so on button release this will set reloop_toggle to 0 instead of beatloop_activate. Try this:
```
    if (value) {
        if (engine.getValue(group, "loop_enabled")) {
            engine.setValue(group, "reloop_toggle", true);
        } else {
            engine.setValue(group, "beatloop_activate", true);
            engine.setValue(group, "beatloop_activate", false);
        }
    }
```